docs/source/guides/upload.md

```diff
@@ -371,11 +376,89 @@ In addition to [`upload_file`] and [`upload_folder`], the following functions al
 
 For more detailed information, take a look at the [`HfApi`] reference.
 
-## Push files with Git LFS
+## Tips and tricks for large uploads
+
+The above guide shows how to technically upload files to the Hub. However there are a few things to know when you're
+trying to upload a large quantity of data. Given the time it takes to stream the data, it can be very annoying to
+get a commit to fail at the end of the process. We gathered a list of tips and recommendations to think about before
+starting the upload.
+
+### Technical limitations
+
+What are we talking about when we say "large uploads" and what are their associated limitations? Large uploads can be
+very diverse, from repositories with a few huge files (e.g. model weights) to repositories with thousands of small files
+(e.g. an image dataset). Each repository will have its own limitations so here is a list of factors to consider:
+
+- **Repository size**: The total size of the data you're planning to upload. There is no hard limit on the size of a repository on
+Hub. However if you plan to upload hundreds of GBs or even TBs of data, we would be grateful if you could let us know
```

Comment:
"let us know in advance" -> how ? maybe suggest the discord server for example

Reply:
Or email

Reply:
@osanseviero which email address would you suggest? Maybe datasets@huggingface.co ?

Reply:
yes datasets sounds good to me (website@ also an option)